Added new folder and files for Vault License FAQs (#11809)
hashicorp/vault: website/content/docs/enterprise/license/faqs.mdx

---
layout: docs
page_title: Frequently Asked Questions (FAQs)
description: An overview of license.
---

# Frequently Asked Questions (FAQs)

This FAQs section is for the license changes introduced in Vault Enterprise 1.8. We plan to deprecate the old license system (vault write sys/license) where autoload will be the only way to manage your license.

## Q: When will these licensing changes be released for Vault?

> The licensing and EULA changes will be available with Vault 1.8-RC1 on June 16, 2021. Vault 1.8 will be GA on July 28, 2021.

Reviewer: @aartiiyengar  Are we comfortable giving a specific GA date (July 28)?

```suggestion
The licensing and end-user license agreement (EULA) changes will be available with Vault 1.8-RC1 on June 16, 2021. Vault 1.8 will become generally available on July 28, 2021.
```

## Q: Will these license changes impact HCP Vault?

No, these changes will not impact HCP Vault.

## Q: Do these license changes impact all Vault customers/licenses?

| Customer/licenses                                                                  | Impacted? |
| ---------------------------------------------------------------------------------- | --------- |

> | ENT binaries (Evaluation or non-Evaluation downloaded from releases.hashicorp.com) | Yes       |

Reviewer: ```suggestion
| ENT binaries (Evaluation or non-Evaluation downloaded from [releases.hashicorp.com](https://releases.hashicorp.com/vault/)) | Yes       |
```

| Open-Source (OSS)                                                                  | No        |

> | Baked-in license binaries (pro/premium)                                            | No        |

Reviewer: @aartiiyengar Correct me if I'm wrong, but based on the [Vault Enterprise Packaging](https://www.datocms-assets.com/2885/1621618395-vault-enterprise-packaging-fy22.pdf), the available packages are **Standard**, **Plus**, **Premium** and **ADP**.  

```suggestion
| Baked-in license binaries (plus/premium)                                            | No        |
```

## Q: What is the product behavior change introduced by the licensing changes?

With Vault 1.8, a valid license is required on-disk (auto-loading) or in-storage for Vault to boot up successfully.

> - Existing clusters upgrading to Vault 1.8(+) will not be affected. Customers may continue to use an in-storage license for their existing clusters until they decide to move to auto-loading. Please see the question below: [Is there a limited time for support of licenses that are in storage?](#q-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage).

Reviewer: ```suggestion
- Existing clusters upgrading to Vault 1.8 or later will not be affected. Customers may continue to use an in-storage license for their existing clusters until they decide to move to auto-loading. Please see the question below: [Is there a limited time for support of licenses that are in storage?](#q-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage).
```

> - Existing clusters upgrading to Vault 1.8(+) will not be affected. Customers may continue to use an in-storage license for their existing clusters until they decide to move to auto-loading. Please see the question below: [Is there a limited time for support of licenses that are in storage?](#q:-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage?).

Reviewer: ```suggestion
- Existing clusters upgrading to Vault 1.8(+) will not be affected. Customers may continue to use an in-storage license for their existing clusters until they decide to move to auto-loading. Please see the question below: [Is there a limited time for support of licenses that are in storage?](#q-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage).
```

> - When deploying new Vault 1.8(+) clusters, please ensure a valid license exists on-disk (auto-loading).

Reviewer: ```suggestion
- When deploying new Vault 1.8 or later clusters, please ensure a valid license exists on-disk (auto-loading).
```

## Q: How will Vault behave at startup when a license expires or terminates?

> ß

Reviewer: ```suggestion

```

> When a license expires, Vault continues to function until the license terminates. This behavior exists today and remains unchanged in Vault 1.8. The grace period, defined as the time between license expiration and license termination, is 1 day for EVAL licenses (as of 1.8), and ten years for non-Evaluation licenses. Customers must install a valid license before the grace period expires. This license can be autoloaded or loaded into storage using existing methods (`/sys/license`) depending on the applicable scenarios as described above.

Reviewer: ```suggestion
When a license expires, Vault continues to function until the license terminates. This behavior exists today and remains unchanged in Vault 1.8. The grace period, defined as the time between license expiration and license termination, is 1 day for evaluation licenses (as of 1.8), and ten years for non-evaluation licenses. Customers must install a valid license before the grace period expires. This license can be autoloaded or loaded into storage using existing methods (`/sys/license`) depending on the applicable scenarios as described above.
```

When license terminates (upon grace period expiry), Vault will seal itself and customers will need a valid license in order to successfully bring-up Vault. If a valid license was not installed after license expiry, customers will need to install one, and this license will need to be autoloaded.

## Q: What is the impact on EVAL licenses due to this change?

The 6-hour trial period for Enterprise binaries allows these particular binaries to run without a license file and will be deprecated as of Vault 1.8(+). This change means that any clusters deployed with Vault 1.8(+) ENT binaries must have a valid license, and the license must be on the disk (auto-loading).

## Q: Are there any changes to existing methods for manual license loading ( API or CLI)?

There are no significant changes to the API or CLI license loading with Vault 1.8.
However, the `/sys/license` endpoint is slated for deprecation in a future release.

> Please see the question: [Is there a limited time for support of licenses that are in storage?](#q:-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage?).

Reviewer: ```suggestion
Please see the question: [Is there a limited time for support of licenses that are in storage?](#q-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage).
```

> ## Q: Is there a grace period when Eval licenses expire?

Reviewer: ```suggestion
## Q: Is there a grace period when evaluation licenses expire?
```

Evaluation licenses have a 1-day grace period. The grace period is the time until the license expires. Upon expiration, Vault will seal and will require a valid license to unseal and function properly.

## Does this affect clients?

> No, there is no impact to the Vault agent.

Reviewer: ```suggestion
No, there is no impact to the Vault Agent.
```

## Q: Are the license files locked to a specific cluster?

The changes to licenses apply to all nodes in a cluster. The license files are not locked to a cluster, but are independently applied to the appropriate clusters.

## Q: Will a EULA check happen every time a Vault restarts?

> Yes, starting with Vault 1.8, ENT binaries will be subjected to a EULA check. The release of Vault 1.8 introduces the EULA check for Evaluation licenses (non-Evaluation licenses are evaluated with a EULA check during contractual agreement) .

Reviewer: ```suggestion
Yes, starting with Vault 1.8, ENT binaries will be subjected to a EULA check. The release of Vault 1.8 introduces the EULA check for evaluation licenses (non-evaluation licenses are evaluated with a EULA check during contractual agreement) .
```

Although the agreement to a EULA occurs only once (when the user receives their license), Vault will check for the presence of a valid license every time a node is restarted.

> When a customer upgrades existing clusters to a 1.8(+) release, a valid license must exist to upgrade successfully. This valid license must be in storage or on-disk (auto-loaded). Please see the below question: [Is there a limited time for support of licenses that are in storage?](#q:-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage?).

Reviewer: ```suggestion
When a customer upgrades existing clusters to a 1.8(+) release, a valid license must exist to upgrade successfully. This valid license must be in storage or on-disk (auto-loaded). Please see the below question: [Is there a limited time for support of licenses that are in storage?](#q-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage).
```

Also, when customers deploy new clusters to a 1.8(+) release, a valid license must exist to upgrade successfully. This valid license must be on-disk (auto-loaded).

## Q: What scenarios should a customer plan for due to these license changes?

- **New Cluster Deployment**: When a customer deploys new clusters to a 1.8(+) release, a valid license must exist to successfully deploy. This valid license must be on-disk (auto-loaded).

> - **Eventual Migration**: Since in-storage licenses will not be supported forever, migrating to auto-loaded licenses will be required. Please see the below question: [Is there a limited time for support of licenses that are in storage?](#q:-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage?).

Reviewer: ```suggestion
- **Eventual Migration**: Since in-storage licenses will not be supported forever, migrating to auto-loaded licenses will be required. Please see the below question: [Is there a limited time for support of licenses that are in storage?](#q-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage).
```

## Q: What is the migration path for customers who want to migrate from their existing license-as-applied-via-the-CLI flow to the license on disk flow?

If a Vault 1.7 cluster is upgraded to Vault 1.8(+) and still has a stored license, the operator must migrate using an auto-loaded license.

1. Use the new command `vault license get -signed` to retrieve the license from the storage in their running cluster.

> 2. Put the license on disk, configure auto-loading (new license_path config option, OR new environment variable `VAULT_LICENSE_PATH` OR new environment variable `VAULT_LICENSE` containing the actual license itself).

Reviewer: ```suggestion
2. Put the license on disk, configure auto-loading (new `license_path` config option, OR new environment variable `VAULT_LICENSE_PATH` OR new environment variable `VAULT_LICENSE` containing the actual license itself).
```

3. Restart the Vault node.
4. A warning will explain that the license is in storage and using auto-loading.

> 5. Use the new command `delete /sys/license` to fully transition to auto-loading.

Reviewer: ```suggestion
5. Use the new command `vault delete /sys/license` to fully transition to auto-loading.
```

## Q: What is the migration path for customers who wish to migrate from their existing perpetually licensed binaries to the license on disk flow (auto-loading)?

> A Vault customer who is currently using S3 unlocked binaries, upgrading their clusters to Vault 1.8(+), and wishes to move to auto-loading, will need a valid enterprise binary version 1.8(+) and a valid license. The steps are the same as in the above question: [What is the migration path for customers who want to migrate from their existing license-as-applied-via-the-CLI flow to the license on disk flow?](#q:-what-is-the-migration-path-for-customers-who-want-to-migrate-from-their-existing-license-as-applied-via-the-cli-flow-to-the-license-on-disk-flow?) starting at Step 2.

Reviewer: ```suggestion
A Vault customer who is currently using S3 unlocked binaries, upgrading their clusters to Vault 1.8(+), and wishes to move to auto-loading, will need a valid enterprise binary version 1.8(+) and a valid license. The steps are the same as in the above question: [What is the migration path for customers who want to migrate from their existing license-as-applied-via-the-CLI flow to the license on disk flow?](#q-what-is-the-migration-path-for-customers-who-want-to-migrate-from-their-existing-license-as-applied-via-the-cli-flow-to-the-license-on-disk-flow) starting at Step 2.
```

## Q: What is the migration path for customers who want to migrate from their existing perpetually licensed binaries to the current license on storage flow?

> A Vault customer currently using S3 unlocked binaries is upgrading their clusters to a pre-Vault 1.8 version, the customer must retrieve the corresponding enterprise binary version <1.8 and a valid license to load the license in storage (/sys/license). Please see the below question: [Is there a limited time for support of licenses that are in storage?](#q-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage).

Reviewer: You already said, "pre-Vault 1.8 version", so I don't think you need to say "version <1.8". 

```suggestion
A Vault customer currently using S3 unlocked binaries is upgrading their clusters to a pre-Vault 1.8 version, the customer must retrieve the corresponding enterprise binary and a valid license to load the license in storage (`/sys/license`). Please see the below question: [Is there a limited time for support of licenses that are in storage?](#q-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage).
```

> A Vault customer currently using S3 unlocked binaries is upgrading their clusters to a pre-Vault 1.8 version, the customer must retrieve the corresponding enterprise binary version <1.8 and a valid license to load the license in storage (/sys/license). Please see the below question: [Is there a limited time for support of licenses that are in storage?](#q:-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage?).

Reviewer: ```suggestion
A Vault customer currently using S3 unlocked binaries is upgrading their clusters to a pre-Vault 1.8 version, the customer must retrieve the corresponding enterprise binary version <1.8 and a valid license to load the license in storage (/sys/license). Please see the below question: [Is there a limited time for support of licenses that are in storage?](#q-is-there-a-limited-time-for-support-of-licenses-that-are-in-storage).
```

## Q: What is the path for customers who want to downgrade/rollback from Vault 1.8(+) (auto-loading) to a pre- Vault 1.8 (license in storage)?

> The downgrade procedure remains the same for Vault customers who are currently on Vault 1.8(+), have a license installed via auto-loading, and would like to downgrade their cluster to a pre-1.8 version that does not have auto-loading support and only supports license in storage. Please refer to the [upgrade procedures](https://learn.hashicorp.com/tutorials/vault/sop-upgrade?in=vault/standard-procedures) that remind the customers that they must take a snapshot before the upgrade. Customers will need to restore their version from the snapshot, apply the pre-1.8 enterprise `binary/config` version they wish to roll back, and bring up the clusters.

Reviewer: ```suggestion
The downgrade procedure remains the same for Vault customers who are currently on Vault 1.8(+), have a license installed via auto-loading, and would like to downgrade their cluster to a pre-1.8 version that does not have auto-loading support and only supports license in storage. Please refer to the [upgrade procedures](https://learn.hashicorp.com/tutorials/vault/sop-upgrade?in=vault/standard-procedures) that remind the customers that they must take a snapshot before the upgrade. Customers will need to restore their version from the snapshot, apply the pre-1.8 enterprise binary version they wish to roll back, and bring up the clusters.
```

---
layout: docs
page_title: Frequently Asked Questions (FAQs)
description: An overview of license.
---

# Frequently Asked Questions (FAQs)

This FAQs section is for the license changes introduced in Vault Enterprise 1.8. We will no longer support the old license system (`vault write sys/license`), and autoload will be the only way to manage your license.

> - [Q: When will these licensing changes be release for Vault?](#q-when-will-these-licensing-changes-be-released-for-vault)

Reviewer 2: ```suggestion
- [Q: When will these licensing changes be released for Vault?](#q-when-will-these-licensing-changes-be-released-for-vault)
```
